ui/src/app/utils/runtime-presets.ts

```diff
@@ -29,3 +30,26 @@ export const runtimePresets: {
     }
   }
 };
+
+export const applyPresetOverride = (runtime) => {
+  if (!runtime) {
+    return runtime;
+  }
+
+  const newRuntime = {...runtime};
+
+  // if runtime configuration type is a default, override its config with preset values
+  // The Deleted check is so that we only update the user's runtime to the latest preset values
+  // after they delete their runtime (ex. not while its actively in use).
+  if (newRuntime && newRuntime.status === RuntimeStatus.Deleted) {
```

Comment:
IMO this business logic belongs outside of this helper function. I realize this may result in some small duplication, but the separation of responsibilities is clearer if the caller only calls `applyPresetOverride` in cases where it actually wants to apply a presetOverride. Otherwise you probably need a super verbose method name to convey the semantics accurately.